src/src/PluginStructs/P025_data_struct.cpp

```diff
@@ -17,24 +17,55 @@ int16_t P025_data_struct::read() {
   config |= (0x8000); // Start a single conversion
 
   Wire.beginTransmission(i2cAddress);
-  Wire.write((uint8_t)(0x01));
+  Wire.write((uint8_t)(0x01));   // access to config register
   Wire.write((uint8_t)(config >> 8));
   Wire.write((uint8_t)(config & 0xFF));
   Wire.endTransmission();
 
-  delay(9); // See https://github.com/letscontrolit/ESPEasy/issues/3159#issuecomment-660546091
-  return readRegister025(0x00);
+  while (isReady025()==false) delay(1);
+//  delay(9); // See https://github.com/letscontrolit/ESPEasy/issues/3159#issuecomment-660546091
+  return readConversionRegister025();
 }
 
-uint16_t P025_data_struct::readRegister025(uint8_t reg) {
+uint8_t P025_data_struct::getMux()
+{
+  return (mux);
+}
+
+uint16_t P025_data_struct::readConversionRegister025() {
+  uint16_t wConversionRegister=0x0000;
+
+  Wire.beginTransmission(i2cAddress);
+  Wire.write((0x00));    // access to conversion register
+  Wire.endTransmission();
+
+
+  if (Wire.requestFrom(i2cAddress, (uint8_t)2) == 2) {
+    wConversionRegister=Wire.read() << 8;
+    wConversionRegister+=Wire.read();
+  }
+  return wConversionRegister;
+}
+
+
+
+bool P025_data_struct::isReady025() {
+  uint16_t wConfigReg=0x0000;
+
   Wire.beginTransmission(i2cAddress);
-  Wire.write((0x00));
+  Wire.write((0x01));    // access to conversion register
   Wire.endTransmission();
 
-  if (Wire.requestFrom(i2cAddress, (uint8_t)2) != 2) {
-    return 0x8000;
+  if (Wire.requestFrom(i2cAddress, (uint8_t)2) == 2) {
+    wConfigReg=(Wire.read() << 8);
+    wConfigReg+=Wire.read();
+    if ((wConfigReg & 0x8000)==0x8000)  // bit15=0 performing a conversion   =1 not performing a conversion
+      return true;
+    else
+      return false;
```

Comment:
That can be simplified to `return (wConfigReg & 0x8000) == 0x8000;`

Reply:
Or even simpler: `return (wConfigReg & 0x8000);`